adagucserverEC/CLegendRenderers/CCreateLegendRenderDiscreteLegend.cpp

```diff
@@ -129,40 +139,100 @@ int CCreateLegend::renderDiscreteLegend(CDataSource *dataSource, CDrawImage *leg
   /**
    * Defined blocks based on defined interval
    */
-
   if (definedLegendOnShadeClasses) {
-    float blockHeight = float(legendImage->Geo->dHeight - 30) / float(styleConfiguration->shadeIntervals->size());
-    /* Legend classes displayed as blocks in the legend can have a maximum and a minimum height depending on the amount of classes and legendheight */
-    if (blockHeight > 12) blockHeight = 12;
-    if (blockHeight < 3) blockHeight = 3;
-    char szTemp[1024];
-    for (size_t j = 0; j < styleConfiguration->shadeIntervals->size(); j++) {
-      CServerConfig::XMLE_ShadeInterval *s = (*styleConfiguration->shadeIntervals)[j];
-      if (s->attr.min.empty() == false && s->attr.max.empty() == false) {
-        int cY1 = int(cbH - ((j)*blockHeight) * scaling);
-        int cY2 = int(cbH - ((((j + 1) * blockHeight) - 2)) * scaling);
-        CColor color;
-        if (s->attr.fillcolor.empty() == false) {
-          color = CColor(s->attr.fillcolor.c_str());
-        } else {
-          color = legendImage->getColorForIndex(CImageDataWriter::getColorIndexForValue(dataSource, parseFloat(s->attr.min.c_str())));
+    // If this type of legend has too many classes (15+) with a relatively small height (<=500 px), we
+    // will treat it differently to create a summarised version, with the following characteristics:
+    // - The class border is removed for extra visibility
+    // - Labels are simplified and will only show the min of the interval the class represents
+    // - Only one every 5 labels will be printed (this is just a rule of thumb) plus the top label
+    // - Only classes the min and the max data value will be added
+    const size_t MAX_DISCRETE_CLASSES = 15;
+
+    // Case where the legend has too many classes to plot
+    if (styleConfiguration->shadeIntervals->size() > MAX_DISCRETE_CLASSES && legendImage->Geo->dHeight <= 500) {
```

Comment:
Maybe check if this can be set by a flag via RenderSettings or check if it can be deteriminded by blockheight.